### DISCUSSION TITLE
Search not working in a link field

### DISCUSSION THREAD

**Discussion by ARCHES-Academy** [statamic/cms#4591] [Troubleshooting] [answered]
On one of my entries, I have a Entry field which pulls a list of 1000+ entries to select from. The titles aren't super helpful to search through, so I'd like to be able to search the content field as well (and maybe a couple others?).

I've looked at the docs for search, but I can't seem to get it working. Here's what I've tried:

I added a new search index for standards:

``` php
    'indexes' => [

        'default' => [
            'driver' => 'local',
            'searchables' => 'all',
            'fields' => ['title'],
        ],

         'standards' => [
             'driver' => 'local',
             'searchables' => 'collection:standards',
             'fields' => ['title', 'content', 'standards_category'],
         ],

    ],
```

Then I added a reference to the index in the standards yaml file:

``` yaml
title: Standards
search_index: standards
I ran the update search index command:
php please search:update standards
```

After doing this, the field still is not searching anything other than the `title` field. Did I miss a step somewhere?

PS: I also tried editing the original search index, but that didn't do anything either.

**Comment by duncanmcclean** [comment, marked as answer]
Fieldtypes like the Link fieldtype or the Entry fieldtype don't make use of Statamic's search functionality (which is what you're updating the config for).

The code for doing search in the fieldtypes are pretty dumb (with it hard coded to search the entry title) - see: https://github.com/statamic/cms/blob/3.2/src/Fieldtypes/Entries.php#L145

Unfortunatley, that means right now I don't _think_ there's a way to get what you're trying to do to work.

**Reply by duncanmcclean** [reply]
In fact, you might be able to get away with overriding the Entries method and use your own `getIndexQuery` method to search through the other column as well but I don't have enough time to give it a go and not sure if it would cause ideas elsewhere.

**Reply by ARCHES-Academy** [reply]
Well, that's unfortunate, but thanks for taking the time to help me out. I'll give that a try when I get the time, and I'll let you know here how it goes (for future readers with similar questions).
